exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2407 @@
+package deribit
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+)
+
+// Deribit is the overarching type across this package
+type Deribit struct {
```

Comment:
Please run `/cmd/wrapper_issues` and `wrapper_coverage` tools.

Reply:
My test has passed the wrapper_issues test. but, when I run wrapper_coverage it keeps exiting with a nil pointer dereference panic message on other exchanges while calling AuthenticateWebsocket() method. I think the context.Context parameter passed to the method is nil.

<img width="875" alt="Screenshot 2023-01-10 at 11 35 48 PM" src="https://user-images.githubusercontent.com/39623015/211656763-26324c69-9376-419d-b2d3-c370a93a0072.png">

Reply:
Potential fix [incoming](https://github.com/thrasher-corp/gocryptotrader/pull/1119)